tools/c7n_kube/c7n_kube/actions.py

```diff
@@ -45,3 +49,80 @@ def process_resource_set(self, client, model, resources):
         for r in resources:
             log.info('%s %s' % (op_name, r))
         pass
+
+
+class PatchAction(MethodAction):
+    """
+    Patches a resource
+
+    Requires patch and namespaced attributes on the resource definition
+    """
+    def validate(self):
+        if not self.manager.get_model().patch:
+            raise PolicyValidationError('patch attribute not defined for resource')
+        return self
+
+    def get_permissions(self):
+        patch = self.manager.get_model().patch
+        return ''.join([a.capitalize() for a in patch.split('_')])
+
+    def patch_resources(self, op, patch_args, namespaced, resources):
+        for r in resources:
+            patch_args['name'] = r['metadata']['name']
```

Comment:
rather than mutating the dict from the caller, make a separate copy for the method to use and mutate.